extensions/eclipse-che-theia-plugin-ext/src/browser/plugin/che-plugin-frontend-service.ts

```diff
@@ -0,0 +1,146 @@
+/********************************************************************************
+ * Copyright (C) 2019 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, inject } from 'inversify';
+import {
+    HostedPluginServer,
+    PluginMetadata
+} from '@theia/plugin-ext/lib/common/plugin-protocol';
+
+import {
+    ChePluginMetadata
+} from '../../common/che-protocol';
+
+@injectable()
+export class ChePluginFrontentService {
+
+    @inject(HostedPluginServer)
+    protected readonly hostedPluginServer: HostedPluginServer;
+
+    async getDeployedPlugins(filter: string): Promise<ChePluginMetadata[]> {
+        if (this.hasType(filter, '@installed')) {
+            let pluginList = await this.getAllDeployedPlugins();
+            pluginList = this.filter(pluginList, filter);
+            return pluginList;
+        }
+
+        return [];
+    }
+
+    // @installed
+    // @builtin
+    // @enabled
+    // @disabled
+    hasType(filter: string, type: string) {
```

Comment:
Minor: apply, please, return type `:boolean`. But up to you.

Reply:
Done. Thanks